libraries/AP_Compass/AP_Compass_QMC5883P.cpp

```diff
@@ -0,0 +1,224 @@
+//
+// Down sampling Rate OSR2
+#define QMC5883P_OSR2_8	0x08
+
+//RNG
+#define QMC5883P_RNG_30G (0x00 << 2)
+#define QMC5883P_RNG_12G (0x01 << 2)
+#define QMC5883P_RNG_8G  (0x10 << 2)
+#define QMC5883P_RNG_2G  (0x11 << 2)
+
+#define QMC5883P_SET_XYZ_SIGN 0x29
+//Reset
+#define QMC5883P_RST 0x80
+
+//Status Val
+#define QMC5883P_STATUS_DATA_READY 0x01
+
+extern const AP_HAL::HAL &hal;
+
+AP_Compass_Backend *AP_Compass_QMC5883P::probe(AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev,
+        bool force_external,
+        enum Rotation rotation)
+{
+    if (!dev) {
+        return nullptr;
+    }
+    AP_Compass_QMC5883P *sensor = new AP_Compass_QMC5883P(std::move(dev),force_external,rotation);
+    if (!sensor || !sensor->init()) {
+        delete sensor;
+        return nullptr;
+    }
+    return sensor;
+}
+
+AP_Compass_QMC5883P::AP_Compass_QMC5883P(AP_HAL::OwnPtr<AP_HAL::Device> dev,
+        bool force_external,
+        enum Rotation rotation)
+    : _dev(std::move(dev))
+    , _rotation(rotation)
+    , _force_external(force_external)
+{
+}
+
+bool AP_Compass_QMC5883P::init()
+{
+    _dev->get_semaphore()->take_blocking();
+
+    _dev->set_retries(10);
+#if 0
+    _dump_registers();
+#endif
+    if (!_check_whoami()) {
+        goto fail;
+    }
+    //As mentioned in the Datasheet 7.2 to do continues mode 0x29 will set sign for X,Y,Z
+    if (!_dev->write_register(QMC5883P_REG_DATA_OUTPUT_Z_MSB, QMC5883P_SET_XYZ_SIGN)||
+        !_dev->write_register(QMC5883P_REG_CONF1,
+                              QMC5883P_MODE_CONTINUOUS|
+                              QMC5883P_ODR_100HZ|
+                              QMC5883P_OSR1_8|
+                              QMC5883P_OSR2_8)||
+        !_dev->write_register(QMC5883P_REG_CONF2,QMC5883P_OSR2_8)) {
+        goto fail;
+    }
+
+    // lower retries for run
+    _dev->set_retries(3);
+
+    _dev->get_semaphore()->give();
+
+    //register compass instance
+    _dev->set_device_type(DEVTYPE_QMC5883P);
+    if (!register_compass(_dev->get_bus_id(), _instance)) {
+        return false;
+    }
+    set_dev_id(_instance, _dev->get_bus_id());
+
+    printf("%s found on bus %u id %u address 0x%02x\n", name,
+           _dev->bus_num(), _dev->get_bus_id(), _dev->get_bus_address());
+
+    set_rotation(_instance, _rotation);
+
+    if (_force_external) {
+        set_external(_instance, true);
+    }
+
+    //Enable 100HZ
+    _dev->register_periodic_callback(10000,
+                                     FUNCTOR_BIND_MEMBER(&AP_Compass_QMC5883P::timer, void));
+
+    return true;
+
+fail:
+    _dev->get_semaphore()->give();
+    return false;
+}
+bool AP_Compass_QMC5883P::_check_whoami()
+{
+    uint8_t whoami;
+    //Affected by other devices,must read registers 0x00 once or reset,after can read the ID registers reliably
+    //    depreciated from L series to P
+    /**    _dev->read_registers(0x00,&whoami,1);
+    if (!_dev->read_registers(0x0C, &whoami,1)||
+      		whoami != 0x01){
+    	return false;
+    }**/
```

Comment:
Remove the dead code now that we know we dont need it.

```suggestion
    //Affected by other devices,must read registers 0x00 once or reset,after can read the ID registers reliably
    //    depreciated from L series to P
    /**    _dev->read_registers(0x00,&whoami,1);
    if (!_dev->read_registers(0x0C, &whoami,1)||
      		whoami != 0x01){
    	return false;
    }**/
```

Reply:
thanks Removed the Dead code